### DISCUSSION TITLE
Is there a way to toggle only between "asc" and "desc" sort without the third option "undefined"?

### DISCUSSION THREAD

**Discussion by blerrgh** [bryanmylee/svelte-headless-table#77] [Ideas]
I'd like to make a table where clicking the headers only switches between ascending and descending sorts and never "clears."  Is there an easy way to do that?

I don't mind doing something like 

`on:click={() props.sort.order = (props.sort.order === 'asc') ? 'desc' : 'asc' }`

But it might be even better to have an option that can passed into the plugin options

```
sort: addSortBy({
         ...
	toggleClearEnable: false
});
```

Or something similar?

**Comment by bryanmylee** [comment]
I can add that as a configuration option for the plugin, but one question that pops up is what should the default state be before any interactions are enabled?

Should it act as usual where there is no sorting before interactions?

**Reply by blerrgh** [reply]
I'd probably expect it to still have no sorting by default (unless initialSortKeys is passed as an option).  Then toggle between asc and desc when the row heading is clicked.  

Could also be nice to have a way to clear sorting globally programmatically.  If my understanding is correct right now I'd need to iterate through all the columns and call WritableSortKeys.clearId() or headerCell.clear()? Might be nice to have a global clear() function somewhere to remove sorting from all columns.

Thanks for your help and quick response!

**Reply by bryanmylee** [reply]
I should have made this clearer in the documentation, but [sortKeys](https://svelte-headless-table.bryanmylee.com/docs/plugins/add-sort-by#sortkeys-writablesortkeys) is basically a writable store. You can set it to `[]` to clear all sorting.

```ts
const {pluginStates} = table.createViewModel(columns);
const {sortKeys} = pluginStates.sort;

// clear sorting
$sortKeys = [];
```

**Reply by blerrgh** [reply]
Oh right, perfect. I'm brand new to Svelte so forgive my silly questions.

Thanks!

**Comment by bryanmylee** [comment]
@blerrgh This should now be addressed with #82 in v0.15.3.

Simply customize the `toggleOrder` property.

```ts
...
sort: addSortBy({
  toggleOrder: ['asc', 'desc'],
}),
...
```

**Reply by blerrgh** [reply]
Just updated and it works great. Thanks for addressing this so quickly!
